shared/provision/code-page/index.js

```diff
@@ -0,0 +1,479 @@
+      <QRImage code={props.textCode} cellSize={5} />
+    </Box2>
+  ) : (
+    <Box2
+      style={collapseStyles([
+        styles.qrContainer,
+        props.currentDeviceAlreadyProvisioned && styles.qrContainerFlip,
+      ])}
+      direction="vertical"
+    >
+      <Box2 direction="vertical" style={styles.qrImageContainer}>
+        <QRImage code={props.textCode} />
+      </Box2>
+      <QRScan />
+    </Box2>
+  )
+
+class EnterText extends React.Component<Props, {code: string}> {
+  state = {code: ''}
+
+  _submit = () => {
+    this.props.onSubmitTextCode(this.state.code)
+  }
+
+  render() {
+    return (
+      <Box2 direction="vertical" style={styles.enterTextContainer} gap="small">
+        <PlainInput
+          autoFocus={true}
+          multiline={true}
+          onChangeText={code => this.setState({code})}
+          onEnterKeyDown={this._submit}
+          rowsMin={3}
+          placeholder={`Type the ${this.props.otherDeviceType === 'mobile' ? '9' : '8'}-word secret code`}
+          textType="Terminal"
+          style={styles.enterTextInput}
+          value={this.state.code}
+        />
+        <WaitingButton
+          fullWidth={true}
+          type="PrimaryColoredBackground"
+          backgroundMode="Green"
+          label="Submit"
+          onClick={this._submit}
+          style={styles.enterTextButton}
+          waitingKey={Constants.waitingKey}
+        />
+      </Box2>
+    )
+  }
+}
+
+const ViewText = (props: Props) => (
+  <Box2 direction="vertical" style={styles.viewTextContainer}>
+    <Text type="Terminal" style={styles.viewTextCode}>
+      {props.textCode}
+    </Text>
+  </Box2>
+)
+
+const Instructions = (p: Props) => (
+  <Box2 direction="vertical">
+    {p.currentDeviceAlreadyProvisioned ? (
+      <React.Fragment>
+        <Text type="Header" style={styles.instructions}>
+          Ready to provision using
+        </Text>
+        <Text type="Header" style={styles.instructionsItalic}>
+          {p.currentDeviceName}.
+        </Text>
+      </React.Fragment>
+    ) : (
+      <React.Fragment>
+        <Text type="Header" style={styles.instructions}>
+          In
+          <Text type="Header" style={styles.instructionsItalic}>
+            {' '}
+            {p.otherDeviceName}
+          </Text>
+          , go to
+        </Text>
+        <Text type="Header" style={styles.instructions}>
+          Devices
+          <Text type="Header" style={styles.instructionsCarets}>
+            {` ${String.fromCharCode(iconMeta['iconfont-arrow-right'].charCode || 0)} `}
+          </Text>
+          <Text type="Header" style={styles.instructions}>
+            Add new
+          </Text>
+          <Text type="Header" style={styles.instructionsCarets}>
+            {` ${String.fromCharCode(iconMeta['iconfont-arrow-right'].charCode || 0)} `}
+          </Text>
+          <Text type="Header" style={styles.instructions}>
+            New {p.currentDeviceType === 'desktop' ? 'computer' : 'mobile'}.
```

Comment:
I think "phone" instead of "mobile", according to Zeplin

Reply:
fixed in part 4: https://github.com/keybase/client/pull/12922/commits/5a4d3342e235129996bc11e24166a286e2604061